docs/tutorial/writing-tests/test-plan.rst

```diff
@@ -0,0 +1,383 @@
+use a regex to apply the override (to match a subset of a template expansion)
+or when the job you want to override is not in the list due to ``nested-parts``
+(that we will introduce further below).
+
+Bootstrap Inclusions
+====================
+
+As we have previously discussed, resources are the backbone of Checkbox
+information gathering. Using the data they generate jobs are skipped or ran and
+templates are instantiated. Although Checkbox does try to pull all resources
+and dependencies you may need into a test plan automatically, jobs may
+interfere or break resources so, ideally, we would like to run them before
+anything else. Bootstrap include does exactly this.
+
+The bootstrap section of a test plan is the initial information gathering phase
+of a test plan. Although there aren't any limitations as to what you can include
+in the ``bootstrap_include`` section, we advise to only put there information
+gathering jobs.
+
+Let's go back to our test plan and move the resource job ``network_iface_info``
+in the ``bootstrap_include`` section:
+
+.. code-block::
+
+  unit: test plan
+  id: tutorial-extended
+  _name: Extended Tutorial Test Plan
+  bootstrap-include:
+    network_iface_info
+  include:
+    network_available
+    network_speed certification-status=blocker
+  certification_status_overrides:
+    apply blocker to network_available
+
+You may have noticed we weren't including ``network_available_interface`` in
+the test plan before, this is because it would not have expanded
+deterministically. One of the dangers of letting Checkbox
+automatically pull resource jobs for you is that, in some situations, like
+``template-resource``, it won't do it. If you were to remove the test that
+actually pulled the resource automatically (the one that uses it as in the
+``resource`` field), you would inadvertently lose test coverage.
+
+Let's update the test plan including it:
+
+.. code-block::
+
+  unit: test plan
+  id: tutorial-extended
+  _name: Extended Tutorial Test Plan
+  bootstrap-include:
+    network_iface_info
+  include:
+    network_available_interface
+    network_available
+    network_speed certification-status=blocker
+  certification_status_overrides:
+    apply blocker to network_available
+
+When we run expand on the test plan, two important changes occur in the output:
+
+First, the resource job is no longer visible - this is expected! The bootstrap
+section of a test plan is meant to gather essential data before the main test
+execution but is not composed of actual tests, so the jobs there are excluded
+from the expand command.
+
+Second, our newly added template wasn't expanded. This happens because a
+template is expanded on the result of a resource, and only running the resource
+can give that output (that is often specific to one machine!). If we want to
+see what would the test plan expand to on the current machine, we can use
+``list-bootstrapped``:
+
+Try to run the following
```

Comment:
```suggestion
When we run ``expand`` on the test plan, two important changes occur in the
output:

- First, the resource job is no longer visible – this is expected! The
  bootstrap section of a test plan is meant to gather essential data before the
  main test execution but is not composed of actual tests, so the jobs there
  are excluded from the expand command.
- Second, our newly added template wasn't expanded. This happens because a
  template is expanded on the result of a resource, and only running the
  resource can give that output (that is often specific to one machine!). If we
  want to see all the jobs that would be executed on the current machine if we
  ran that test plan, we can use ``list-bootstrapped``:
```